### ISSUE TITLE
[Bug] Embedding strings trigger false-positives in hint-minified-js

### ISSUE THREAD

**Issue by antross** [webhintio/hint#2922]
While investigating some other issues with this hint, I determined that the token ratio used in it's calculation is thrown off by embedded strings which can be single tokens of significant length. Notably on a few sites such as bing.com this has been embedded SVG. I'm sure we could run into similar issues with other embedded types of code.

I propose we fix this by updating the calculation to exclude string tokens. Then subtract their length from the total length before calculating the ratio. This will likely require re-calculating what the best target ratio is, though. Initial tests suggest 75 (even 85) is too permissive with this change.

**Comment by antross** [comment]
```ts


        const getImprovementIndex = ({ sourceCode, tokens }: ScriptParse) => {
            let stringTokenCount = 0;
            let stringTokenLength = 0;

            for (const token of tokens) {
                if (token.type.label === 'string') {
                    stringTokenCount++;
                    stringTokenLength += token.value.length;
                }
            }

            const contentLength = sourceCode.length;
            const tokenRatio = (tokens.length - stringTokenCount) / (contentLength - stringTokenLength);

            return Math.round((1 - tokenRatio) * 100);
        };
```

**Comment by antross** [comment]
Also @molant, do you recall if there was a reason we didn't just check to see if the content of the script is all/mostly on one line? While such a simple approach could be tricked, but in practical use it may be more accurate.

**Comment by molant** [comment]
When using libraries and frameworks there are usually licenses at the top so there are more than 1 line so it wasn't as straightforward 😟

**Comment by antross** [comment]
That's true, but should be easy to strip comments before testing line length (or average line length). Just brainstorming here to see if another approach might prove more stable. I'm skeptical of the `tokenRatio`.

**Comment by molant** [comment]
I have faith in the `tokenRatio` approach.

This file is from bing.com for the vendors:
https://business.bing.com/msb/r/vendors_main_feed_f8e37b105e7ebee1f4c5.js

it has multiple lines with multiple comments in the middle and with JS starting at the end of copyright notices in the same line. Not saying we can't come up with something, but we will have to analyze a lot of different files and edge cases.
Also we will have to take into account template string that have multiple lines which is the original problem...

**Comment by antross** [comment]
Ok, we'll continue to refine the original approach then. Just wanted to make sure we weren't missing out on a simpler opportunity 😊

**Comment by molant** [comment]
The other option I could think of is instead of "minifying" do the exact opposite: "prettify". If there are no changes in the number of lines (or minimum) then the source is not minified. Don't know how complicated it will be to implement that or if there's a (lightweight) package that could do that.
